### PR TITLE
Radio button filter

### PR DESCRIPTION
**Pull request by ChanJit**
** Please provide as much detail as possible, but feel free to remove irrelevant sections **

Adding a radio button in style guid

BREAKING CHANGE:

no breaking change

RFC URL:

no API Change in this pull request

MIGRATION GUIDE:

Before:

there is new component so no update in this section

After:

Radio button been added

EXAMPLE USAGE:

import {
  Radio
} from 'seek-asia-style-guide/react';

<Radio
  buttonFormat
  id="myRadio"
  inputProps={{
    checked: false,
    onChange: () => {...}
  }}
  name="stillInRole"
/>

Screen Shot:

Normal radio button format
radio button in radio format

Button Format
radio button in button format

### CODE REVIEW

File: react/Radio/Radio.demo.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  initialProps: {
+    name: 'stillInRole',
+    // Documentation only:
+    id: 'myRadio',
+    inputProps: {
+      checked: false,
+      onChange: () => {}
+    }
+  },
+  options: [
```
Comment (mengtzu) [inline react/Radio/Radio.demo.js]: Might be good to do multiple buttons for the demo ala `ButtonGroup`

File: react/Radio/Radio.demo.less [hunk 2]
```diff
@@ -0,0 +1,8 @@
+.root {
+  display: inline-block;
+  flex-direction: column;
+
+  > :global(*) {
+    min-width: 150px; // :(
```
Comment (mengtzu) [inline react/Radio/Radio.demo.less]: It's a demo, code quality is less important here.  Just as long as it doesn't go into production 😅

**Comment by mengtzu** [comment]
Surge preview: http://seek-asia-style-guide--28d05572caf8cffe6a85831ebd6fa9f1c4e70d13.surge.sh/radio

File: react/Radio/Radio.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+export default class Radio extends Component {
+  static displayName = 'Radio';
+
+  static propTypes = {
+    id: PropTypes.string.isRequired,
+    className: PropTypes.string,
+    label: PropTypes.string,
+    labelProps: PropTypes.object,
+    inputProps: PropTypes.object,
+    buttonFormat: PropTypes.string
```
Comment (mengtzu) [inline react/Radio/Radio.js]: I think `type` or `component` is a bit more consistent with existing components (see `Button` and `Checkbox`)

File: react/Radio/Radio.less [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+.labelText {
+  .input:checked + .label & {
+    font-weight: @sk-bold;
+  }
+}
+
+.buttonText {
+  .input:checked + .label & {
+    font-weight: @sk-bold;
+    color: @sk-gray-light;
```
Comment (mengtzu) [inline react/Radio/Radio.less]: Should replace `@sk-` colours with SEEK Asia colours where applicable.

**Comment by mengtzu** [comment]
@ChanJit thanks for cleaning up a bunch of stuff while you were in there!

Have you showed this to the designers yet to check they're happy with it?  Use the surge link above if you can't easily show them via Zoom or whatever :)

File: react/Radio/Radio.js [hunk 5]
```diff
@@ -0,0 +1,90 @@
+import styles from './Radio.less';
+
+import PropTypes from 'prop-types';
+
+import React, { Component } from 'react';
+import classnames from 'classnames';
+import Text from '../Text/Text';
+
+function combineClassNames(props = {}, ...classNames) {
```
Comment (PepeFranco) [inline react/Radio/Radio.js]: I have seen this function declared and used in several components. Should it be something that is out it an common file?

**Comment by mengtzu** [comment]
@ChanJit is this still going ahead?

**Comment by nicolassenechal** [comment]
@ChanJit still going forward or should we close for now?

**Comment by ChanJit** [comment]
closing the pull request, because is no longer needed

**Pull request closed without merging**